src/main/java/name/valery1707/kaitai/KaitaiMojo.java

```diff
@@ -179,51 +185,12 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 		project.addCompileSourceRoot(generatedRoot.normalize().toFile().getAbsolutePath());
 	}
 
-	static URL prepareUrl(URL url, String version) throws MojoExecutionException {
-		if (url == null) {
-			try {
-				url = new URL(format(URL_FORMAT, version, version));
-			} catch (MalformedURLException e) {
-				throw new MojoExecutionException("Invalid version: " + version, e);
-			}
-		}
-		return url;
-	}
-
-	static Path prepareCache(File target, MavenSession session, Log log) throws MojoExecutionException {
-		Path cache;
-		if (target == null) {
-			Path repository = new File(session.getLocalRepository().getBasedir()).toPath();
-			cache = repository.resolve(".cache").resolve("kaitai").normalize();
+	private Path detectCacheDir() {
+		if (cacheDir != null) {
+			return cacheDir.toPath();
```

Comment:
\[Checkstyle\] WARNING: 'if' child has incorrect indentation level 24, expected level should be 6\.